app/models/customer.rb

```diff
@@ -0,0 +1,22 @@
+class Customer < ActiveRecord::Base
+  ACCOUNT_TYPES = %w(Free Basic Premium)
+
+  # A customer has one log for every time they've called in. If the
+  # customer is deleted, their logs are also deleted.
+  # (Data retention? What's that?)
+  has_many :logs, dependent: :destroy
+
+  validates :account_type, inclusion: {in: ACCOUNT_TYPES}
+  validates :name, length: {in: 1..255}
+  validates :email, length: {in: 1..255}
+
+  def number_of_calls
+    logs.count
```

Comment:
denormalize this by adding it to a column on the customer's table, every time they make a new call add 1 to that column for them.